src/p2p/mod.rs

```diff
@@ -17,13 +17,20 @@ mod transport;
 pub use addr::{MultiaddrWithPeerId, MultiaddrWithoutPeerId};
 pub use {behaviour::KadResult, swarm::Connection};
 
+/// An IPFS swarm.
 pub type TSwarm<T> = Swarm<behaviour::Behaviour<T>>;
 
+/// Defines the configuration for an IPFS swarm.
 pub struct SwarmOptions {
+    /// The keypair for the PKI based identity of the local node.
     pub keypair: Keypair,
+    /// The peer address of the local node created from the keypair.
     pub peer_id: PeerId,
+    /// The peers to connect to on startup.
     pub bootstrap: Vec<(Multiaddr, PeerId)>,
+    /// Enables mdns for peer discovery and announcement when true.
     pub mdns: bool,
+    /// Custom Kademlia protocol name.
```

Comment:
```suggestion
    /// Custom Kademlia protocol name, see [`IpfsOptions::kad_protocol`].
```
I didn't check if that'll be a good link however linking to the longer description with the default in `IpfsOptions` seems like the way to go. Another way would be to link to `IpfsOptions` at the type level.